File: SenD.client/src/services/MonstersService.js
import { AppState } from "../AppState.js"
import { Monster } from "../models/Monster.js"
import { logger } from "../utils/Logger.js"
import { api } from "./AxiosService.js"

class MonstersService {
  // Create Monster
  async createMonster(monsterData) {
    const res = await api.get(`api/monsters`, monsterData)
    const monster = new Monster(res.data)
    logger.log('CREATE MONSTER', monster)

    AppState.monsters.unshift(monster)
  }
  // Get Monsters
  async getMonsters() {
    const res = await api.get('api/monsters')
    const monsters = res.data.map(m => new Monster(m))
    logger.log('GET MONSTERS', monsters)
    AppState.monsters = monsters
  }
  // Delete/Deprecate Monster
  // Update Monster
}
export const monstersService = new MonstersService()

File: SenD/Services/MonstersService.cs
namespace SenD.Services;

public class MonstersService
{
  private readonly MonstersRepository _monstersRepository;

  public MonstersService(MonstersRepository monstersRepository)
  {
    _monstersRepository = monstersRepository;
  }

  internal Monster createMonster(Monster monsterData)
  {
    int monsterId = _monstersRepository.createMonster(monsterData);
    Monster monster = getMonsterById(monsterId);
    return monster;
  }

  internal Monster getMonsterById(int monsterId)
  {
    Monster monster = _monstersRepository.getMonsterById(monsterId);
    if (monster == null)
    {
      throw new Exception($"Bad Monster ID: {monsterId}");
    }
    return monster;
  }

  internal List<Monster> getMonsters()
  {
    List<Monster> monsters = _monstersRepository.getMonsters();
    return monsters;
  }
}